cmd/ddev/cmd/flags.go

```diff
@@ -0,0 +1,357 @@
+		*v = defValueValue(strconv.FormatInt(0, 10))
+	case ftTest3: // used for testing only
+		*v = ""
+	default:
+		if implemented := ValidTypes[typ]; implemented {
+			// Mandatory implementation missing -> panic
+			panic(fmt.Sprintf("Error implementation of DefValue validation missing for type '%s'", typ))
+		}
+	}
+
+	return nil
+}
+
+func (v *noOptDefValValue) validate() error {
+	return nil
+}
+
+func (v *annotationsValue) validate() error {
+	return nil
+}
+
+func (f *Flag) validateFlag(longOptions *map[nameValue]bool, shortOptions *map[shorthandValue]nameValue) error {
+	errors := ""
+
+	// Chech all fields
+	errors += extractError(f.Name.validate(longOptions))
+	errors += extractError(f.Shorthand.validate(shortOptions, f.Name))
+	errors += extractError(f.Usage.validate())
+	errors += extractError(f.Type.validate())
+	errors += extractError(f.DefValue.validate(f.Type))
+	errors += extractError(f.NoOptDefVal.validate())
+	errors += extractError(f.Annotations.validate())
+
+	if errors != "" {
+		return fmt.Errorf("%s", errors)
+	}
+
+	return nil
+}
+
+// Init initializes the Flags structure.
+func (f *Flags) Init(commandName, script string) {
+	f.CommandName = commandName
+	f.Script = script
+}
+
+func (f *Flags) validateFlags(flags *FlagsDefinition) error {
+	var errors string
+
+	// Temporay vars to precheck for duplicated flags. It's still possible
+	// other commands will introduce the same flags which is tested
+	// afterwards by cobra.
+	long := map[nameValue]bool{"help": true}
+	short := map[shorthandValue]nameValue{"h": "help"}
+
+	for i := range *flags {
+		flag := &(*flags)[i]
+
+		// Additional validations of the flag fields
+		flagErrors := extractError(flag.validateFlag(&long, &short))
+
+		if flagErrors != "" {
+			errors += extractError(formatErrorItem(1, "*", "for flag '%s':%s", flag.Name, flagErrors))
+		}
+	}
+
+	if errors != "" {
+		return fmt.Errorf("The following problems were found in the flags definition of the command '%s' in '%s':%s", f.CommandName, f.Script, errors)
+	}
+
+	return nil
+}
+
+// LoadFromJSON imports the defs provided by the custom command as json into
+// the flags structure.
+func (f *Flags) LoadFromJSON(data string) error {
+	if data == "" {
+		return nil
+	}
+
+	var err error
+	var defs = FlagsDefinition{}
+
+	// Import the JSON to the FlagsDefinition structure and return in case of
+	// error
+	if err = json.Unmarshal([]byte(data), &defs); err != nil {
+		return err
+	}
+
+	// Validate the user provided flags and return in case of error
+	if err = f.validateFlags(&defs); err != nil {
+		return err
+	}
+
+	// Assign the data to the field
+	f.Definition = defs
+	return nil
+}
+
+// AssignToCommand iterates the flags and assigns it to the provided command.
+func (f *Flags) AssignToCommand(command *cobra.Command) error {
+	var errors string
+
+	for _, flag := range f.Definition {
+		// Create the flag at the command
+		switch flag.Type {
```

Comment:
As you can see from the list above there are many more types which could be implemented in the future if needed. For now, I've focused on the most important.